core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -32,6 +50,7 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
 
     sealed class ResolveResult {
```

Comment:
Okay, say there that resolver has failed with an exception (reports an error and returns ScriptDependencies.Empty). Should we update script highlighting in IDE in that case? We should display an error for sure but is marking all references as unresolved actually helpful? I believe that API you propose doesn't allow to handle this scenario.